compiler/fir/java/src/org/jetbrains/kotlin/fir/java/JavaUtils.kt

```diff
@@ -80,9 +82,23 @@ internal fun FirTypeRef.toConeKotlinTypeProbablyFlexible(
 
 internal fun JavaType.toFirJavaTypeRef(session: FirSession, javaTypeParameterStack: JavaTypeParameterStack): FirJavaTypeRef {
     val annotations = (this as? JavaClassifierType)?.annotations.orEmpty()
+    val type = this
     return buildJavaTypeRef {
         annotations.mapTo(this.annotations) { it.toFirAnnotationCall(session, javaTypeParameterStack) }
-        type = this@toFirJavaTypeRef
+        this.type = type
```

Comment:
I think this code is not needed at all. I'll investigate why `FirJavaTypeRef` remains in resolved FIR.